bin/misc-functions.sh

```diff
@@ -383,12 +383,13 @@ preinst_mask() {
 		fi
 	done
 
-	install_mask "${ED}" "${INSTALL_MASK}"
-
-	# remove share dir if unnessesary
-	if has nodoc $FEATURES || has noman $FEATURES || has noinfo $FEATURES; then
-		rmdir "${ED%/}/usr/share" &> /dev/null
-	fi
+	# Store the final value of INSTALL_MASK in build-info
+	local x
+	set -f
+	local IFS=$' \t\n\r'
+	x=$(echo ${INSTALL_MASK})
+	[[ -n $x ]] && echo "$x" > "${PORTAGE_BUILDDIR}"/build-info/INSTALL_MASK
```

Comment:
If INSTALL_MASK is empty, we can do like for DEFINED_PHASES here, and put `-` in the file. Then the unmerge code will know whether or not to fall back to `self.settings['INSTALL_MASK']` here (which we don't want to do if it could create orphans).

Reply:
I don't think there's a point in putting `-` when we can just print an empty string.